library/std/src/thread/mod.rs

```diff
@@ -1076,6 +1078,27 @@ impl ThreadId {
         }
     }
 
+    #[cfg(target_has_atomic = "64")]
+    fn new() -> ThreadId {
+        use crate::sync::atomic::AtomicU64;
+
+        static COUNTER: AtomicU64 = AtomicU64::new(1);
+        const COUNTER_MAX: u64 = 1u64 << 63;
+
+        let id = COUNTER.fetch_add(1, SeqCst);
```

Comment:
There is no need for synchronization, so relaxed ordering would be fine.

Reply:
I used `SeqCst` because it was already imported :sweat_smile: